### PR TITLE
RND203-324: Attach ath

### PR DESCRIPTION
**Pull request by nzulinski-mobica**

### CODE REVIEW

File: sys/dev/ath/ath_hal/ah_internal.h [hunk 1]
```diff
@@ -639,6 +639,7 @@ extern	void *ath_hal_malloc(size_t);
 extern	void ath_hal_free(void *);
 
 /* common debugging interfaces */
+#define AH_DEBUG
```
Comment (Rafal-Litwinski-Mobica) [inline sys/dev/ath/ath_hal/ah_internal.h]: Probably AH_DEBUG shouldn't be enabled on our main branch

Comment (nzulinski-mobica) [reply sys/dev/ath/ath_hal/ah_internal.h]: Done

File: sys/dev/ath/ath_hal/ar9002/ar9271_attach.c [hunk 2]
```diff
@@ -203,6 +203,7 @@ ar9271Attach(uint16_t devid, HAL_SOFTC sc,
 	ahp->ah_maxTxTrigLev		= MAX_TX_FIFO_THRESHOLD >> 1;
 
 	AH_PRIVATE(ah)->ah_isusb	= TRUE;
+	AH_PRIVATE(ah)->ah_ispcie	= FALSE;
```
Comment (Mikolaj-Filar-Mobica) [inline sys/dev/ath/ath_hal/ar9002/ar9271_attach.c]: It's overwritten on line 233.

File: sys/dev/ath/if_ath_usb.c [hunk 3]
```diff
@@ -434,13 +434,17 @@ static int
...
-	struct ath_usb_softc *usc =  device_get_softc(self);
-	struct ath_softc *sc = &usc->sc_sc;
+	struct ath_softc *sc = device_get_softc(self);
 	struct ieee80211com *ic = &sc->sc_ic;
+	struct ath_usb_softc *usc;
 	usb_error_t error;
 	struct usb_endpoint *ep, *ep_end;
 
 	device_set_usb_desc(self);
+	usc = malloc(sizeof(struct ath_usb_softc), M_TEMP,
```
Comment (Mikolaj-Filar-Mobica) [inline sys/dev/ath/if_ath_usb.c:444]: Is the memory freed? 

I also wonder if there is a memory leak if we replace usc allocated by device_get by other structure, but let's think about it other time :)

File: sys/dev/ath/if_ath_usb.c [hunk 4]
```diff
@@ -457,9 +461,14 @@ ath_usb_attach(device_t self)
 	sc->ops.write = ath_usb_write;
 	sc->ops.write_barrier = ath_usb_write_barrier;
 #endif
-
 	mtx_init(&sc->sc_mtx, device_get_nameunit(self), MTX_NETWORK_LOCK, MTX_DEF);
 
+	// ATH_LOCK_INIT(sc);
```
Comment (Mikolaj-Filar-Mobica) [inline sys/dev/ath/if_ath_usb.c:466]: Are these lines needed?

File: sys/dev/ath/ath_hal/ar9002/ar9271_attach.c (hunk 2 above)
Comment (nzulinski-mobica) [reply sys/dev/ath/ath_hal/ar9002/ar9271_attach.c]: yeah I will change it

File: sys/dev/ath/if_ath_usb.c (hunk 4 above)
Comment (nzulinski-mobica) [reply sys/dev/ath/if_ath_usb.c:466]: It is to investigate mutexes so I will leave it

File: sys/dev/ath/if_ath_usb.c (hunk 3 above)
Comment (nzulinski-mobica) [reply sys/dev/ath/if_ath_usb.c:444]: It is freed now and also I changed the size of structure 
`static driver_t ath_usb_driver = {
	.name = "if_ath_usb",
	.methods = ath_usb_methods,
	.size = sizeof(struct ath_softc)
};`

**Review by Mikolaj-Filar-Mobica — APPROVED** [review]

**Review by Rafal-Litwinski-Mobica — APPROVED** [review]

**Pull request merged**